experimental/segmenter/src/provider/lstm.rs

```diff
@@ -110,11 +110,10 @@ pub enum ModelType {
 }
 
 /// The struct that stores a LSTM model.
-#[icu_provider::data_struct(LstmDataV1Marker = "segmenter/lstm@1")]
-#[derive(PartialEq, Debug, Clone)]
+#[derive(PartialEq, Debug, Clone, yoke::Yokeable, zerofrom::ZeroFrom)]
 #[cfg_attr(feature = "datagen", derive(serde::Serialize))]
 #[yoke(prove_covariance_manually)]
-pub struct LstmDataV1<'data> {
+pub struct LstmDataFloat32<'data> {
```

Comment:
I made this change even smaller than the original one so that we can avoid bikeshedding.

Reply:
I saw. There's also the added complexity of now having `LstmMatrix3Float32` 😆